python/cuml/svm/svm_base.pyx

```diff
@@ -72,22 +74,41 @@ cdef extern from "cuml/svm/svm_parameter.h" namespace "ML::SVM":
         double epsilon
         SvmType svmType
 
+cdef extern from "cuml/matrix/matrix.h" namespace "MLCommon::Matrix":
+    cdef cppclass Matrix[math_t]:
+        bool is_dense()
+        DenseMatrix[math_t]* as_dense()
+        CsrMatrix[math_t]* as_csr()
+
+    cdef cppclass CsrMatrix[math_t](Matrix[math_t]):
+        int get_nnz();
+        int* get_indptr();
+        int* get_indices();
+        math_t* get_data();
+        CsrMatrix(int* indptr, int* indices, math_t* data, int nnz, int rows, int cols) except +
+
+    cdef cppclass DenseMatrix[math_t](Matrix[math_t]):
+        math_t* get_data();
+        bool is_row_major();
+        DenseMatrix(math_t* data, int rows, int cols) except +
+  
+
 cdef extern from "cuml/svm/svm_model.h" namespace "ML::SVM":
     cdef cppclass SvmModel[math_t]:
         # parameters of a fitted model
         int n_support
         int n_cols
         math_t b
         math_t *dual_coefs
-        math_t *x_support
+        Matrix[math_t] *support_matrix
```

Comment:
One way to do this cleanly for sparse inputs would be to have the `sv*FitXSparse` compute the indices of the support vectors and then a call can be made to a helper function to copy the rows with the support indices into a new matrix. It should be trivial to compute the sparsity given a list of row indices.